### PR TITLE
cpu/stm32f1: make use of CPU_LINE and STM32_FLASHSIZE

### PR DESCRIPTION
**Pull request by vincent-d**
### Contribution description

Refactor stm32f1 to use CPU_LINE_* and STM32_FLASHSIZE

### Issues/PRs references

#9096

### CODE REVIEW

File: cpu/stm32f1/include/cpu_conf.h [hunk 1]
```diff
@@ -52,19 +48,13 @@ extern "C" {
...
 #endif
 
-#if defined(CPU_MODEL_STM32F103C8)
-#define FLASHPAGE_NUMOF     (64U)
-#elif defined(CPU_MODEL_STM32F103CB) || defined(CPU_MODEL_STM32F103RB)
-#define FLASHPAGE_NUMOF     (128U)
-#elif defined(CPU_MODEL_STM32F103RE) || defined(CPU_MODEL_STM32F103ZE)
-#define FLASHPAGE_NUMOF     (256U)
-#endif
+#define FLASHPAGE_NUMOF     (STM32_FLASHSIZE / FLASHPAGE_SIZE)
```
Comment (aabadie) [inline cpu/stm32f1/include/cpu_conf.h]: Same comment about the moving this change in a separate commit.

**Comment by vincent-d** [comment]
Comment addressed, also remove an enum from vendor header which conflicts with a macro in ccn-lite.

**Comment by vincent-d** [comment]
Squashed and rebased
